app/gui/src/controller/searcher.rs

```diff
@@ -1592,14 +1665,34 @@ pub mod test {
         test.run_until_stalled();
         let list = searcher.actions().list().unwrap().to_action_vec();
         // There are 8 entries, because: 2 were returned from `completion` method, two are mocked,
-        // and all of these are repeasted in "All Search Result" category.
+        // and all of these are repeated in "All Search Result" category.
         assert_eq!(list.len(), 8);
         assert_eq!(list[2], Action::Suggestion(action::Suggestion::FromDatabase(entry1)));
         assert_eq!(list[3], Action::Suggestion(action::Suggestion::FromDatabase(entry9)));
         let notification = subscriber.next().boxed_local().expect_ready();
         assert_eq!(notification, Some(Notification::NewActionList));
     }
 
+    #[wasm_bindgen_test]
+    fn loading_components() {
+        let Fixture { mut test, searcher, entry1, entry9, .. } =
+            Fixture::new_custom(|data, client| {
+                // Entry with id 99999 does not exist, so only two actions from suggestions db
+                // should be displayed in searcher.
+                data.expect_completion(client, None, None, &[1, 99999, 9]);
+            });
+        searcher.reload_list();
+        test.run_until_stalled();
+        let components = searcher.components();
+        if let [module_group] = &components.top_modules()[..] {
+            assert_eq!(module_group.name, entry1.module.to_string());
+            let entries = module_group.entries.borrow();
+            assert_matches!(entries.as_slice(), [e1, e2] if e1.suggestion.name == entry1.name && e2.suggestion.name == entry9.name);
+        } else {
+            ipanic!("Wrong top modules in Component List: {components.top_modules():?}");
```

Comment:
I would never stop reading this function as "I panic". Always when I see the name I'm feeling worried/disturbed :P